ts/private/ts_project_worker.js

```diff
@@ -101,10 +101,22 @@ function createFilesystemTree(root, inputs) {
             }
             notifyWatchers(dirs, parts.base, TYPE.SYMLINK, EVENT_TYPE.ADDED);
         } else if (parts.base) {
-            node[parts.base] = {
-                [Type]: TYPE.FILE
+            const new_node = {
+                [Type]: TYPE.FILE,
             }
-            notifyWatchers(dirs, parts.base, TYPE.FILE, EVENT_TYPE.ADDED);
+            try {
+                const realpath = fs.readlinkSync(path.join(root, p));
```

Comment:
`const stat = fs.lstatSync` will be faster here and you can check `stat.is_symlink()`